presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchColumn.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch;
+
+import com.facebook.presto.spi.type.Type;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+public final class ElasticsearchColumn
+{
+    private final String name;
+    private final Type type;
+    private final String jsonPath;
+    private final String jsonType;
+    private final boolean isList;
+
+    @JsonCreator
+    public ElasticsearchColumn(
+            @JsonProperty("name") String name,
+            @JsonProperty("type") Type type,
+            @JsonProperty("jsonPath") String jsonPath,
+            @JsonProperty("jsonType") String jsonType,
+            @JsonProperty("isList") boolean isList)
+    {
+        checkArgument(!isNullOrEmpty(name), "name is null or is empty");
+        this.name = name;
+        this.type = requireNonNull(type, "type is null");
+        this.jsonPath = requireNonNull(jsonPath, "jsonPath is null");
+        this.jsonType = requireNonNull(jsonType, "jsonType is null");
+        this.isList = requireNonNull(isList, "isList is null");
```

Comment:
`isList` is a primitive, so you don't need `requireNonNull` for it,

Reply:
will fix